### PR TITLE
Fixing Issues with ECR Removal & SSM Connectivity

### PR DESCRIPTION
**Pull request by andrew-aiken**
#### Overview of Changes
- ECR repository will be deleted even with images
  - https://github.com/RhinoSecurityLabs/cloudgoat/issues/200
- Add egress security group on EC2 instance to allow communication with SSM
  - https://github.com/RhinoSecurityLabs/cloudgoat/issues/212
  - https://github.com/RhinoSecurityLabs/cloudgoat/issues/200

#### Testing
- Verify SSM works by default
- ECR removes with images

### CODE REVIEW

File: scenarios/cicd/terraform/sdlc.tf [hunk 1]
```diff
@@ -56,5 +56,6 @@ BASH
 resource "aws_ecr_repository" "app" {
   name                 = local.ecr_repository_name
   image_tag_mutability = "MUTABLE"
+  force_delete         = true
```
Comment (andrew-aiken) [inline scenarios/cicd/terraform/sdlc.tf:59]: When the repository is deleted remove all images

File: scenarios/cicd/terraform/dev_machine.tf [hunk 2]
```diff
@@ -54,14 +54,37 @@ resource "aws_instance" "dev" {
...
+    aws_security_group.allow_egress.id
+  ]
 
   tags = {
     Name        = "dev-instance",
     Environment = "dev"
   }
 }
 
+resource "aws_security_group" "allow_egress" {
```
Comment (andrew-aiken) [inline scenarios/cicd/terraform/dev_machine.tf:69]: Security group to allow egress traffic

File: scenarios/cicd/terraform/dev_machine.tf [hunk 3]
```diff
@@ -54,14 +54,37 @@ resource "aws_instance" "dev" {
   instance_type        = "t3.micro"
   iam_instance_profile = aws_iam_instance_profile.dev.name
   user_data            = templatefile("${path.module}/../assets/dev-machine/provision.sh", { private_ssh_key = tls_private_key.ssh_key.private_key_pem })
-  subnet_id = module.vpc.private_subnets[0]
+  subnet_id            = module.vpc.private_subnets[0]
+
+  security_groups = [
+    aws_security_group.allow_egress.id
```
Comment (andrew-aiken) [inline scenarios/cicd/terraform/dev_machine.tf:60]: Add SG to instance to allow communication with AWS SSM

File: scenarios/cicd/terraform/dev_machine.tf [hunk 4]
```diff
@@ -54,14 +54,37 @@ resource "aws_instance" "dev" {
   instance_type        = "t3.micro"
   iam_instance_profile = aws_iam_instance_profile.dev.name
   user_data            = templatefile("${path.module}/../assets/dev-machine/provision.sh", { private_ssh_key = tls_private_key.ssh_key.private_key_pem })
-  subnet_id = module.vpc.private_subnets[0]
```
Comment (jdearmas) [inline scenarios/cicd/terraform/dev_machine.tf:57]: Formatting

**Review by jdearmas — APPROVED** [review]

**Pull request merged**